mitodl/edx-platform
# Reimplemented legacy dashboard enrollment data functionality

#### What are the relevant tickets?

Closes [mitocw#74](https://github.com/mitocw/edx-platform/issues/74)
⚠️  REBASED ON #18 ⚠️ 

#### What's this PR do?

Gives instructors the ability to see what students are enrolled in specific sections of a course in the remote gradebook

#### Where should the reviewer start?

`lms/templates/instructor/instructor_dashboard_2/remote_gradebook.html` and  `lms/djangoapps/instructor/views/api.py`

#### How should this be manually tested?

- Make sure you have staff access for your test course
- Set the `ENABLE_INSTRUCTOR_REMOTE_GRADEBOOK_CONTROLS` feature to `true` (in `lms.env.json`)
- Create a `REMOTE_GRADEBOOK` setting in `lms.env.json`. This should be a dict with 2 values: 
    ```
    "REMOTE_GRADEBOOK": {
        "URL": "http://lmodproxyqa-mitodl.rhcloud.com/edx_grades",
        "DEFAULT_NAME": "STELLAR:/project/mitxdemosite"
    }
    ```

- Create `REMOTE_GRADEBOOK_USER` and `REMOTE_GRADEBOOK_PASSWORD` settings in `lms.auth.json`. Ask me or Peter for these values
- Navigate to the `Remote Gradebook` tab for some course (eg: `http://localhost:8000/courses/course-v1:edX+DemoX+Demo_Course/instructor#view-remote_gradebook`)
- Click the buttons in the "Remote gradebook enrollments" section and make sure they work.
